### PR TITLE
Fix global nav styles

### PR DESCRIPTION
**Pull request by kosmydel**
### Details

Fixed issues related to global nav styling:
Update:

- [x] 1. Original issue (sign in button cropped): https://github.com/Expensify/App/issues/29057
- [x] 2. https://github.com/Expensify/App/issues/29133
- [x] 3. https://github.com/Expensify/App/issues/29162
- [x] 4. https://github.com/Expensify/App/issues/29085
- [x] 5. https://github.com/Expensify/App/issues/29058 (fixed in [this PR](https://github.com/Expensify/App/pull/28323/files))
- [x] 6.  Change LHP width to 425p ([Slack](https://swmansion.slack.com/archives/C05NM77EUTS/p1697115011699729))
- [x] 7. https://github.com/Expensify/App/issues/29079 
- [x] 8. Attachment background: https://github.com/Expensify/App/issues/29057#issuecomment-1763985970

### Fixed Issues

$ https://github.com/Expensify/App/issues/29057
PROPOSAL: N/A

### Tests

- [ ] Verify that no errors appear in the JS console
- [ ] Check each fixed issue from the above list if it is fixed.
- [ ] 1. Sign in button on Global Nav is cropped (the original issue)
  - [ ] Create a public room
  - [ ] Copy link to it
  - [ ] Sign out
  - [ ] Paste the link
  - [ ] Verify the avatar design in the top left corner
- [ ] 2. Set the status on your profile (make sure you have permissions). Check if the settings button in the top-left corner is correctly cropped.
- [ ] 3. Traffic lights button on macOS desktop
  - [ ] Run the desktop app and log in.
  - [ ] Check if the avatar (in the top-left corner) looks good
- [ ] 4. Check if the Settings Background for profile details is correct.
- [ ] 5. Fixed in another PR
- [ ] 6. **[IMPORTANT]** Check LHP sizing on each platform.
  - [ ] Resize the screen. Check if the LHP and RHP new widths work correctly.
  - [ ] Send money request. Check if the resizing works correctly.
  - [ ] Log out from the app. Check the sizing of the login screen.
  - [ ] Go to the report. Press + button to open the ReportActionCompose menu. Check if it is positioned correctly (left margin).
  - [ ] Go to the not found (`http://localhost:8082/not-found`) page. Check if the margins are set correctly.
  - [ ] Open modal RHP docked. Go to Profile settings, upload a new avatar, and check if the avatar edit photo modal is displayed correctly.
  - [ ] Make more tests related to changing the width of the LHP as it is really tricky change.
- [ ] 7. Check if the border around avatars is the same as the background in the LHP report list.
  - [ ] Check multiple avatars background (i.g. in the room)
  - [ ] Check workspace + avatar background
- [ ] 8. Verify colors:
  - [ ] Send an attachment. Verify background color.
  - [ ] Verify colors of activity indicator.

### Offline tests

N/A

### QA Steps

- [ ] Verify that no errors appear in the JS console
- [ ] Do the checklist from the above.
- [ ] Focus on the new width of the LHP, as this might cause regressions. Try different screens and window widths.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

android - native

**OLD:**

Screenshot 2023-10-16 at 15 50 27
Screenshot 2023-10-16 at 15 50 22

</details>

<details>
<summary>Android: mWeb Chrome</summary>

android - web

**OLD:**

Screenshot 2023-10-16 at 16 07 06
Screenshot 2023-10-16 at 16 07 00

</details>

<details>
<summary>iOS: Native</summary>

ios - native

**OLD:**

Screenshot 2023-10-16 at 15 08 35
Screenshot 2023-10-16 at 15 08 31

</details>

<details>
<summary>iOS: mWeb Safari</summary>

ios - web

**OLD:**

Screenshot 2023-10-16 at 15 45 18
Screenshot 2023-10-16 at 15 45 13

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

web

**OLD:**

Screenshot 2023-10-16 at 14 55 59

Screenshot 2023-10-16 at 15 01 46

</details>

<details>
<summary>MacOS: Desktop</summary>

desktop

</details>

### CODE REVIEW

**Comment by robertKozik** [comment]
Hi, I cannot edit the details of this PR as @kosmydel created it. Until than, here is how the fixed headers looks like. CC. @shawnborton 

<details>
<summary>Web</summary>

image


</details>

<details>
<summary>Mobile Web - Chrome</summary>

image


</details>

<details>
<summary>Mobile Web - Safari</summary>

image


</details>

<details>
<summary>Desktop</summary>

image


</details>

<details>
<summary>iOS</summary>

image


</details>

<details>
<summary>Android</summary>

image


</details>

**Comment by shawnborton** [comment]
iOS still looks broken. We should keep it basically how it currently is, but just add the left global nav color behind the floating island/status bar.

**Comment by robertKozik** [comment]
I missed that message on slack, my bad. So something like this one?
image

**Comment by shawnborton** [comment]
Perfect!

**Comment by hayata-suenaga** [comment]
changing the assignment of C+ per [this discussion](https://expensify.slack.com/archives/C02NK2DQWUX/p1698149578288879)

File: src/components/HeaderGap/index.desktop.js [hunk 1]
```diff
@@ -2,21 +2,33 @@ import React, {PureComponent} from 'react';
...
 import styles from '../../styles/styles';
+import stylePropTypes from '../../styles/stylePropTypes';
 
 const propTypes = {
     /** Styles to apply to the HeaderGap */
-    // eslint-disable-next-line react/forbid-prop-types
-    styles: PropTypes.arrayOf(PropTypes.object),
+    styles: stylePropTypes,
+
+    /** If the HeaderGap is placed in the LHP */
```
Comment (situchan) [inline src/components/HeaderGap/index.desktop.js:11]: We're more familiar with LHN than LHP.

Is this correct?
Screenshot 2023-10-25 at 6 19 37 PM

Comment (stitesExpensify) [reply src/components/HeaderGap/index.desktop.js:11]: FYI the terminology we have decided on now has changed again 😅

**Comment by robertKozik** [comment]
Branch got merged with main and is ready for review.

**Comment by situchan** [comment]
Can we also rename all variables? I don't see any occurrences of `lhp` on main

**Comment by robertKozik** [comment]
Sure on it

**Comment by robertKozik** [comment]
done 👌🏼  @situchan

File: src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js [hunk 2]
```diff
@@ -38,6 +42,8 @@ function BaseSidebarScreen(props) {
                         <SubNavigation
                             onLinkClick={startTimer}
                             insets={insets}
+                            // Don't display radius if platform is desktop. In this cases HeaderGap component is handling it.
+                            shouldDisplayRadius={!isDesktop}
```
Comment (stitesExpensify) [inline src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:46]: Why does this only affect desktop and not web?

**Review by stitesExpensify — COMMENTED** [review]
Looking good, just a couple of comments

File: src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js (hunk 2 above)
Comment (robertKozik) [reply src/pages/home/sidebar/SidebarScreen/BaseSidebarScreen.js:46]: In desktop we want the radius to 'hug' the top of the window and at the same time we don't want clickable components to be overlapped by the top bar of the window. To achieve it we used the `HeaderGap` component which is rendered above the `Sidebar` screen component. If we won't disable the radius here we would end up with two radius one over the another.
On other platforms, there is no need to use the `HeaderGap` to achieve this. Using the `HeaderGap` solution on them as well would end up with excessive padding on the top of the sidebar.

**Comment by situchan** [comment]
@robertKozik please fix conflict when you get a chance

File: src/components/HeaderGap/index.desktop.js [hunk 3]
```diff
@@ -2,21 +2,33 @@ import React, {PureComponent} from 'react';
...
+    isSidebar: PropTypes.bool,
 };
 
 class HeaderGap extends PureComponent {
     render() {
-        return <View style={[styles.headerGap, ...this.props.styles]} />;
+        if (!this.props.isSidebar) {
```
[the remark this answers is not in the document]
Comment (robertKozik) [reply src/components/HeaderGap/index.desktop.js]: Done 👌🏼

**Comment by robertKozik** [comment]
@situchan main merged

**Comment by hayata-suenaga** [comment]
asked @situchan if he can review this PR on slack

**Review by hayata-suenaga — DISMISSED** [review]
I was only able to glance over the code but it looks good to me as long as @situchan's test goes well 🙇

**Comment by robertKozik** [comment]
resolved conflicts 👌🏼

**Comment by situchan** [comment]
> asked @situchan if he can review this PR on slack

Maybe you asked on C+ channel but all C+ don't have access after slack migration.
I can complete review by tomorrow

**Comment by robertKozik** [comment]
@hayata-suenaga Can we close this PR/issue per your [slack message](https://swmansion.slack.com/archives/C05NM77EUTS/p1698879161209599)? It's related to the PR which is going to be reverted anyways

**Comment by situchan** [comment]
@robertKozik can you please redirect that message here? I don't have any context.

**Comment by hayata-suenaga** [comment]
> @hayata-suenaga Can we close this PR/issue per your [slack message](https://swmansion.slack.com/archives/C05NM77EUTS/p1698879161209599)? It's related to the PR which is going to be reverted anyways

Yes I'm closing this issue now 🙇

**Comment by hayata-suenaga** [comment]
@situchan I assigned you to the issue for payment

**Pull request closed without merging**